### PR TITLE
#715-metrics: added metrics

### PR DESCRIPTION
**Pull request by otbe**

### CODE REVIEW

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/TimedOperation.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/RegistryMetricsImpl.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/MetricEvent.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcher.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5245358&bid=17191718)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/MetricEvent.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/RegistryMetricsImpl.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcher.java  2
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/TimedOperation.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5245358&bid=17191718)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/TimedOperation.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcher.java  2
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/MetricEvent.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/RegistryMetricsImpl.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5245358&bid=17191718)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/TimedOperation.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/MetricEvent.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcher.java  2
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/RegistryMetricsImpl.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5245358&bid=17191718)

File: factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java [hunk 1]
```diff
@@ -42,7 +47,16 @@ public void put(@NonNull Fact f, @NonNull String transformationChainId) {
     public Optional<Fact> find(@NonNull UUID eventId, int version,
             @NonNull String transformationChainId) {
         String key = CacheKey.of(eventId, version, transformationChainId);
-        return Optional.ofNullable(cache.get(key));
+
+        Optional<Fact> result = Optional.ofNullable(cache.get(key));
+
+        if (result.isPresent()) {
```
Comment (uweschaefer) [inline factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java]: isn't a ternary nicer here?

File: factcast-store-pgsql/src/test/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcherTest.java [hunk 2]
```diff
@@ -53,8 +63,12 @@ public void testCreateSchemaUrl() throws Exception {
 
     @Test
     public void testNullContracts() throws Exception {
-        TestHelper.expectNPE(() -> new HttpRegistryFileFetcher(null));
-        TestHelper.expectNPE(() -> new HttpRegistryFileFetcher(null, new OkHttpClient()));
+        val uut = new HttpRegistryFileFetcher(baseUrl, client, registryMetrics);
```
Comment (uweschaefer) [inline factcast-store-pgsql/src/test/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcherTest.java]: wondering, if this is still necessary to do with lombok using Objects.requireNonNull

**Comment by uweschaefer** [comment]
lgtm

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/RegistryMetricsImpl.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcher.java  2
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/TimedOperation.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/MetricEvent.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5245358&bid=17191718)

File: factcast-store-pgsql/src/test/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcherTest.java (hunk 2 above)
Comment (otbe) [reply factcast-store-pgsql/src/test/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcherTest.java]: Yeah I was thinking about the same. Maybe skip for new things 🤔

File: factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java (hunk 1 above)
Comment (otbe) [reply factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java]: done

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/cache/InMemTransformationCache.java  2
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/RegistryMetricsImpl.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/http/HttpRegistryFileFetcher.java  2
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/MetricEvent.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/metrics/TimedOperation.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5245358&bid=17191718)

**Pull request merged**